### DISCUSSION TITLE
Better understanding of PrismaClient location

### DISCUSSION THREAD

**Discussion by snakysnake** [prisma/prisma#22839] [Q&A]
### Question

It is best practice to create the PrismaClient() as few times as possible in your application's flow.
That's why a good practice would be to inject the Prisma Clients to the parts of your application that need it.
One common way to do this in Express.JS Applications would be to pass it using the request object (req).
In my case, I pass the PrismaClient() as req.prisma to functions / endpoints in my app.
Unfortunately, the intellisense never recognizes that my PrismaClient() is now not const prisma but req.prisma.

Is there a way to make the intellisense smarter?
Or is this a skill issue, could I improve my code to tell intellisense that req.prisma is a PrismaClient()?

If anyone knows, let me know.

**Comment by nurul3101** [comment]
Hello @snakysnake 👋 

> It is best practice to create the PrismaClient() as few times as possible in your application's flow.

Yes, it is recommended to have only one PrismaClient Instance. Each PrismaClient instance manages its own connection pool. If there are multiple PrismaClient instances, then your connections will be split. 

> the intellisense never recognizes that my PrismaClient() is now not const prisma but req.prisma.

Could you please share a minimal reproduction so that I can take a look?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
